query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresResultAnalyzer.java

```diff
@@ -37,19 +37,19 @@ static PostgresResultAnalyzer create(
       LinkedHashSet<String> selectedAttributes,
       TableDefinition tableDefinition)
       throws SQLException {
-    Map<String, Integer> logicalNameToPhysicalNameIndex = new HashMap<>();
+    Map<String, Integer> physicalNameColIndexMap = new HashMap<>();
+    ResultSetMetaData metaData = resultSet.getMetaData();
+    for (int colIndex = 0; colIndex < metaData.getColumnCount(); colIndex++) {
+      String physName = metaData.getColumnName(colIndex);
+      physicalNameColIndexMap.put(physName, colIndex);
+    }
 
+    Map<String, Integer> logicalNameToPhysicalNameIndex = new HashMap<>();
     for (String logicalName : selectedAttributes) {
-      String name = tableDefinition.getPhysicalColumnName(logicalName);
-      ResultSetMetaData metaData = resultSet.getMetaData();
-      for (int colIndex = 0; colIndex < metaData.getColumnCount(); colIndex++) {
-        String physName = metaData.getColumnName(colIndex);
-        if (physName.equalsIgnoreCase(name)) {
-          logicalNameToPhysicalNameIndex.put(logicalName, colIndex);
-          break;
-        }
-      }
+      String physName = tableDefinition.getPhysicalColumnName(logicalName);
+      logicalNameToPhysicalNameIndex.put(logicalName, physicalNameColIndexMap.get(physName));
```

Comment:
can `physicalNameColIndexMap.get(physName)` be null? Do we need to check for contains? Ideally, it shouldn't be, right?

Reply:
Shouldn't be given it is based on selections